### PR TITLE
Update for GDAL 3.5

### PR DESCRIPTION
**Pull request by zap150**

### CODE REVIEW

File: geo/gdal.hpp [hunk 1]
```diff
@@ -86,6 +89,23 @@ UTILITY_GENERATE_ENUM_IO_CI(::GDALDataType,
...
+                            ((GDT_Int32)("Int32"))
+                            ((GDT_Float32)("Float32"))
+                            ((GDT_Float64)("Float64"))
+                            ((GDT_CInt16)("CInt16"))
+                            ((GDT_CInt32)("CInt32"))
+                            ((GDT_CFloat32)("CFloat32"))
+                            ((GDT_CFloat64)("CFloat64"))
+                            ((GDT_Unknown)("Unknown"))
+                            ((GDT_TypeCount)("TypeCount"))
+                            )
```
Comment (vaclavblazek) [inline geo/gdal.hpp:107]: Why the 64 bit ints are not added here. You have to duplicate whole code.
i.e.:
```
...
                            ((GDT_Int32)("Int32"))
#if GDAL_VERSION_NUM >= 3050000
                            ((GDT_UInt64)("UInt64"))
                            ((GDT_Int64)("Int64"))
#endif
                            ((GDT_Float32)("Float32"))
                            ((GDT_Float32)("Float32"))
...
```
Once we ditch older GDAL versions we just remove these two preprocessor directives.

Comment (zap150) [reply geo/gdal.hpp:107]: Issues a warning and crashes due to `-Werror`. Cannot put #if into a macro.

Comment (vaclavblazek) [reply geo/gdal.hpp:107]: Aha, make sense. OK

**Review by vaclavblazek — APPROVED** [review]
OK

**Pull request merged**